addons/hw_drivers/iot_handlers/interfaces/DisplayInterface_L.py

```diff
@@ -19,31 +16,51 @@ class DisplayInterface(Interface):
-                power_state_hdmi = Vcgencmd().display_power_state(hdmi_port.get(hdmi))
-                if power_state_hdmi == 'on':
-                    iot_device = {
-                        'identifier': hdmi,
-                        'name': 'Display hdmi ' + str(x_screen),
-                        'x_screen': str(x_screen),
-                    }
-                    display_devices[hdmi] = iot_device
-                    x_screen += 1
-        except subprocess.CalledProcessError:
-            _logger.warning('Vcgencmd "display_power_state" method call failed')
+            # Get xrandr output and parse lines
+            # 'check_output' method returns a bytes object, so we need to decode it to a string and split it by lines
+            # 'split' method is used to get only the last element of each line : 'number of monitors' then 'identifiers'
+            os.environ['DISPLAY'] = ":0"
+            parsed_xrandr_output = [
+                o.split()[-1]
+                for o in subprocess.run(['xrandr', '--listactivemonitors'], check=True, stdout=subprocess.PIPE)
+                .stdout
+                .decode('utf-8')
+                .split('\n')
+                if o
+            ]
+            # Skip the first line which is the number of monitors
+            hdmi_monitors = [
+                name for name in parsed_xrandr_output[1:]
+                if any(substr in name for substr in ['HDMI', 'Composite'])
+            ]
 
-        if not len(display_devices):
-            # No display connected, create "fake" device to be accessed from another computer
-            display_devices['distant_display'] = {
-                'name': "Distant Display",
+            for x_screen, name in enumerate(hdmi_monitors):
+                hdmi = 'hdmi_' + str(x_screen)
+                display_devices[hdmi] = {
+                    'identifier': hdmi,
+                    'name': 'Display - ' + name,
+                    'x_screen': str(x_screen),
+                }
+
+            if not len(display_devices):
+                # No display connected, create "fake" device to be accessed from another computer
+                display_devices['distant_display'] = {
+                    'name': "Distant Display",
+                }
+        except FileNotFoundError:
+            # Versions < 24.06 don't have 'xrandr': subprocess command will then fail with FileNotFound error.
+            # We create a 'fake' display, in order to open a browser anyway
+            display_identifier = 'hdmi_0'
+            display_devices['hdmi_0'] = {
+                'identifier': display_identifier,
+                'name': 'Display - ' + display_identifier,
+                'x_screen': '0',
             }
+        except subprocess.CalledProcessError:
+            # xrandr command will fail on RPI 5 if no device is connected, with error "can't open display :0"
+            # (not the case on RPI 3/4 which create a 'default' display)
```

Comment:
This is also failing for me on the RPi 4 with no display connected - shouldn't we also return a fake display here?